component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/extension/ComponentSchemaEnricher.java

```diff
@@ -47,16 +49,22 @@ public Map<String, String> onComponent(final Type type, final Annotation[] annot
             return emptyMap();
         }
 
-        final Optional<String> fixed = Arrays.stream(annotations)
+        final Optional<FixedSchema> fixed = Arrays.stream(annotations)
                 .filter(a -> a.annotationType().equals(FixedSchema.class))
                 .findFirst()
-                .map(FixedSchema.class::cast)
-                .map(FixedSchema::value);
+                .map(FixedSchema.class::cast);
 
         if (!fixed.isPresent()) {
             return emptyMap();
         }
 
-        return Collections.singletonMap(FIXED_SCHEMA_META_PREFIX, fixed.get());
+        final FixedSchema fixedSchema = fixed.get();
+        final Map<String, String> metadata = new HashMap<>();
+        metadata.put(FIXED_SCHEMA_META_PREFIX, fixedSchema.value());
+        if (fixedSchema.flows().length > 0) {
+            metadata.put(FIXED_SCHEMA_FLOWS_META_PREFIX, String.join(",", fixedSchema.flows()));
```

Comment:
if fixedSchema.flows().length == 0 : metadata.put(FIXED_SCHEMA_FLOWS_META_PREFIX, "__ default __")

Reply:
So the integration doesn't have to manage the implicit case.

Reply:
That's a possible default state indeed.